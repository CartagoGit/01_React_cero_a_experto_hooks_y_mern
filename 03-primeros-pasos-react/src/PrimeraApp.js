// import React, { Fragment } from "react";
import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo: saludoEntrante, subtitulo }) => {
	const saludo = "Hola Mundo";
	const persona = {
		nombre: "Mario",
		edad: 33
	};

	return (
		// <Fragment> se puede usar en vez de <></>
		<>
			<h1> {saludo} </h1>
			<p>Mi primera aplicaci√≥n</p>
			<pre>{JSON.stringify(persona, null, 3)}</pre>
			<p>{saludoEntrante}</p>
			<p>{subtitulo}</p>
		</>
	);
};

PrimeraApp.propTypes = {
	saludo: PropTypes.string.isRequired
};
PrimeraApp.defaultProps = {
	subtitulo: "Soy un subtitulo"
};

export default PrimeraApp;
