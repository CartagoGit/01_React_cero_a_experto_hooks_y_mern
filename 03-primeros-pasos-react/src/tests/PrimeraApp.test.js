// import { render } from "@testing-library/react";

import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import PrimeraApp from "../PrimeraApp";

describe("Pruebas en la primera app de React", () => {
	/*test('Debe mostrar el mensaje "Hola soy Goku"', () => {
		const saludo = "Hola, soy Goku";
		// const wrapper = render(<PrimeraApp saludo={saludo}/>);
        const {getByText} = render(<PrimeraApp saludo={saludo}/>);

        // wrapper.getByText()
        expect (getByText(saludo)).toBeInTheDocument();
	});*/

	test("Deberia mostrar primerapp correctamente", () => {
		const saludo = "Hola, soy Goku";
		const wrapper = shallow(<PrimeraApp saludo={saludo} />);

		expect(wrapper).toMatchSnapshot();
	});

	test("debe enviar el subtitulo enviado por props", () => {
		const saludo = "Hola, soy Goku";
		const subtitulo = "Esto es un subtitulo";
		const wrapper = shallow(
			<PrimeraApp saludo={saludo} subtitulo={subtitulo} />
		);
		const textoParrafo = wrapper.find("p").text();
        // console.log(textoParrafo);
		expect(textoParrafo).toBe(subtitulo);
	});
});
