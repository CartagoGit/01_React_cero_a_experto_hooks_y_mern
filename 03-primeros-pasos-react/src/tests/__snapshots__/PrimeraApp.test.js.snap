// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la primera app de React Deberia mostrar primerapp correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
